src/ITensorNetworks/MPSTensor/GeneralMPSTensor.jl

```diff
@@ -0,0 +1,66 @@
+using ChainRulesCore
+using ZygoteRules: @adjoint
+
+struct GeneralMPSTensor <: AbstractNetwork
+  mps::MPS
+end
+
+function Base.:+(A::GeneralMPSTensor, B::GeneralMPSTensor; kwargs...)
+  if length(A.mps) == 1 && order(A.mps[1]) == 0
+    return GeneralMPSTensor(MPS([ITensor(A[] + B[])]))
+  else
+    return GeneralMPSTensor(+(A.mps, B.mps; kwargs...))
+  end
+end
+
+function Base.getindex(t::GeneralMPSTensor)
+  @assert length(t.mps) == 1
+  @assert order(t.mps[1]) == 0
+  return t.mps[1][]
+end
+
+function ChainRulesCore.rrule(::typeof(getindex), x::GeneralMPSTensor)
+  y = x[]
+  function getindex_pullback(ȳ)
+    x̄ = GeneralMPSTensor(MPS([ITensor(ȳ)]))
+    return (NoTangent(), x̄)
+  end
+  return y, getindex_pullback
+end
+
+"""
+initialization methods
+"""
+Base.isless(a::ITensors.Index, b::ITensors.Index) = a.id < b.id
```

Comment:
If we keep this, we will need a more general definition since the indices could have the same ids but different tags and prime levels. We should define it as something like `id(a) < id(b) && plev(a) < plev(b) && tags(a) < tags(b)`.

Reply:
yes this makes sense

Reply:
Also you should be able to just use `Index` instead of `ITensors.Index`.

Reply:
Done